src/components/common/AccessCheck.jsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import { Redirect, useLocation } from "react-router-dom";
+import { useSelector } from "react-redux";
+import {
+  SHOPPING_CART,
+  SIGN_UP,
+  PROFILE,
+  ADMIN,
+  HOME,
+} from "../../containers/Route.paths.js";
+
+const REDIRECT_TO_HOME = <Redirect to={HOME} />;
+
+const AccessCheck = () => {
+  const { isLoggedIn, isAdmin } = useSelector((state) => state.authentication);
```

Comment:
note that normally as i mentioned for the other PR you don't do this on the clientside (this comment requires no changes)

if i really wanted to i'd simply edit the JS to return that i was an admin so i could access everything here. all permissions checks happen on the server generally

the way to do this would be to use "firebase security rules" using firebase